### PR TITLE
[MetaSchedule] Refactor testing workloads

### PR DESCRIPTION
**Pull request by junrushao1994**

### CODE REVIEW

**Comment by junrushao1994** [comment]
CC: @Hzfengsy @MasterJH5574 @masahi

**Review by Hzfengsy — APPROVED** [review]

File: python/tvm/auto_scheduler/relay_integration.py [hunk 1]
```diff
@@ -482,4 +482,10 @@ def is_auto_scheduler_enabled():
...
         Whether the auto-scheduler is enabled
     """
-    return PassContext.current().config.get("relay.backend.use_auto_scheduler", False)
+    return PassContext.current().config.get(
+        "relay.backend.use_auto_scheduler",
+        False,
+    ) or PassContext.current().config.get(
+        "relay.backend.use_meta_schedule",
+        False,
+    )
```
Comment (MasterJH5574) [inline python/tvm/auto_scheduler/relay_integration.py:491]: Hmmm, why consider meta-schedule in the auto-scheduler?

File: python/tvm/meta_schedule/testing/custom_builder_runner.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+"""Customized builder and runner methods"""
+# pylint: disable=import-outside-toplevel
+
+from typing import TYPE_CHECKING, Dict, List
+
+if TYPE_CHECKING:
+    from tvm.ir import IRModule
+    from tvm.meta_schedule.runner import EvaluatorConfig
+    from tvm.runtime import Device, Module, NDArray
+    from tvm.target import Target
```
Comment (MasterJH5574) [inline python/tvm/meta_schedule/testing/custom_builder_runner.py:26]: Interesting! I never saw this in our previous code. What’s the purpose here?

File: python/tvm/meta_schedule/testing/relay_workload.py [hunk 3]
```diff
@@ -13,156 +13,334 @@
 # "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
-# under the License.
```
Comment (MasterJH5574) [inline python/tvm/meta_schedule/testing/relay_workload.py:16]: Add it back :eyes:

File: python/tvm/relay/build_module.py [hunk 4]
```diff
@@ -287,13 +290,17 @@ def _module_export(module, file_name):  # fcompile, addons, kwargs?
 
 
 @register_func("tvm.relay.build")
+def _build_module_no_factory_impl(mod, target, target_host, params, mod_name):
+    target, target_host = Target.check_and_update_host_consist(target, target_host)
+    return build(mod, target, params=params, mod_name=mod_name).module
+
+
```
Comment (MasterJH5574) [inline python/tvm/relay/build_module.py:297]: I'm curious that, for what consideration we took the implementation out?

**Review by MasterJH5574 — APPROVED** [review]
LGTM! I have some questions and pointed out a nit. Please take a look, thanks :wink: !

File: python/tvm/relay/build_module.py (hunk 4 above)
Comment (junrushao1994) [reply python/tvm/relay/build_module.py:297]: To support kwargs

File: python/tvm/meta_schedule/testing/relay_workload.py (hunk 3 above)
Comment (junrushao1994) [reply python/tvm/meta_schedule/testing/relay_workload.py:16]: Ooops

File: python/tvm/auto_scheduler/relay_integration.py (hunk 1 above)
Comment (junrushao1994) [reply python/tvm/auto_scheduler/relay_integration.py:491]: For operator dispatching

File: python/tvm/meta_schedule/testing/custom_builder_runner.py (hunk 2 above)
Comment (junrushao1994) [reply python/tvm/meta_schedule/testing/custom_builder_runner.py:26]: Yeah it demonstrates that MetaSchedule's builder/runner system is able to handle Relay-level codegen

**Pull request merged**